agent/connect/ca/provider_vault_test.go

```diff
@@ -507,6 +553,138 @@ func TestVaultProvider_Cleanup(t *testing.T) {
 	})
 }
 
+func TestVaultProvider_ConfigureWithAuthMethod(t *testing.T) {
+
+	SkipIfVaultNotPresent(t)
+
+	cases := []struct {
```

Comment:
Curious if we could add some testing for the other vault auth methods as well?

Reply:
Testing other auth methods is a bit tricky since we'd have to stub out the response from whatever the external system vault auth method integrates with, like AWS, GCP or Kubernetes. I'm not sure if there's value in doing that since at that point we're testing vault itself.

Reply:
Yeah I figured that this was the case. 

> that since at that point we're testing vault itself.

That's a fair point -- I think we gain the confidence that auth methods work from the vault testing 🤔